StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthViewController.swift

```diff
@@ -227,7 +246,117 @@ final class PartnerAuthViewController: UIViewController {
             view.addAndPinSubviewToSafeArea(errorView)
         }
     }
+
+    private func handleAuthSessionCompletion(_ status: String, _ authSession: FinancialConnectionsAuthSession) {
+        if status == "success" {
+            self.dataSource.recordAuthSessionEvent(
+                eventName: "success",
+                authSessionId: authSession.id
+            )
+
+            if authSession.isOauthNonOptional {
+                // for OAuth flows, we need to fetch OAuth results
+                self.authorizeAuthSession(authSession)
+            } else {
+                // for legacy flows (non-OAuth), we do not need to fetch OAuth results, or call authorize
+                self.delegate?.partnerAuthViewController(self, didCompleteWithAuthSession: authSession)
+            }
+        } else if status == "failure" {
+            self.dataSource.recordAuthSessionEvent(
+                eventName: "failure",
+                authSessionId: authSession.id
+            )
+
+            // cancel current auth session
+            self.dataSource.cancelPendingAuthSessionIfNeeded()
+
+            // show a terminal error
+            self.showErrorView(
+                FinancialConnectionsSheetError.unknown(
+                    debugDescription: "Shim returned a failure."
+                )
+            )
+        } else { // assume `status == cancel`
+            self.dataSource.recordAuthSessionEvent(
+                eventName: "cancel",
+                authSessionId: authSession.id
+            )
+
+            // cancel current auth session
+            self.dataSource.cancelPendingAuthSessionIfNeeded()
+
+            // whether legacy or OAuth, we always go back
+            // if we got an explicit cancel from backend
+            self.navigateBack()
+        }
+    }
     
+    private func handleAuthSessionCompletionWithNoStatus(_ authSession: FinancialConnectionsAuthSession, _ error: Error?) {
```

Comment:
double checking: the logic of this function didn't change right? just making sure I am not missing double-checking any changed logic

Reply:
yeah just copied over, actually I used xcode's refactor feature :)